platform/staking.md

```diff
@@ -5,35 +5,114 @@ a set of open source libraries and documentation to help blockchain developers t
-
-## Step 2: Describe the staking smart contract
-  - If you staking implementation uses another blockchain like Ethereum, your have to specify the blockchain you're using, the 
-  smart contract address and all inputs necessary to `stake`, `unstake` and `withdraw rewards`. 
-  - Open a pull request to [trust platform](https://github.com/trustwallet/platform) with all necessary information
-  to implement the actions above. 
- 
-## Step 3: Define RPC endpoints for blockatlas
-  - You have to define RPC endpoints to fetch the following data:
-    - Validators list
-    - Staking pool (if applicable)
-    - Current annual APR
-  - After finishing the implementation, open a pull request to [blockatlas](https://github.com/trustwallet/blockatlas) project.
-  You can find an example for Cosmos [here](https://github.com/trustwallet/blockatlas/tree/master/platform/cosmos)
-
-## Step 4: Define Validators Information
-  - Open a pull request to [trust assets](https://github.com/trustwallet/assets) and add all validators
+## Talk about your implementation.
+Create an issue in [Trust Platform](https://github.com/trustwallet/platform/issues) describing your project. Please fill the following questions before proceding with integration:
+
+1) Is your staking on or off-chain?
+   > If it's offchain, please describe how it's implemented
+
+2) If it's on chain, is it implemented natively or using a smart contract? 
+   > Please describe all parameters required to build the message or call a smart contract.
+
+3) What are the staking rules of your blockchain?
+   > Minimum value, locktime, lockout, etc.
+
+4) Do you have any required step before staking? If yes, describe it.
+   > Describe any required sep like token swap or balance freezing
+
+## Return Staking Data
+Trust Platform retrieve staking data from [blockatlas](https://github.com/trustwallet/blockatlas). Blockatlas is used as a common interface for querying data from different blockchains. To support staking you have to implement the following interfaces:
+
+### [Models](https://github.com/trustwallet/blockatlas/blob/master/pkg/blockatlas/staking.go)
+
+##### ValidatorPage
+```go
+type Amount string
+
+type StakingReward struct {
+  Annual float64 `json:"annual"`
+}
+
+type Validator struct {
+  ID            string        `json:"id"`
+  Status        bool          `json:"status"`
+  Reward        StakingReward `json:"reward"`
+  LockTime      int           `json:"locktime"`
+  MinimumAmount Amount        `json:"minimum_amount"`
+}
+
+type ValidatorPage []Validator
+```
+
+##### DelegationsPage
+```go
+type DelegationStatus string
+
+const (
+  DelegationStatusActive  DelegationStatus = "active"
+  DelegationStatusPending DelegationStatus = "pending"
+)
+
+type StakeValidatorInfo struct {
+  Name        string `json:"name"`
+  Description string `json:"description"`
```

Comment:
can you point to go lang code instead? it might change a bit and it will get oudated

Reply:
Ok. I'll remove the models sample and just tell them to check it out